### ISSUE TITLE
Need to characterize attachments with no filename

### ISSUE THREAD

**Issue by hangal** [ePADD/epadd#277]
Sachin's data report says 968 attachments with no filename.

1) Need to understand why this happens and what the attachments are.
2) Should these appear in the listing on the other attachments?

**Comment by chinuhub** [comment]
@jfarwer Did you also observe this issue while ingesting some mbox files? I suspect it happens because of mbox parsing issue(s) which cause empty data to be recognized as an attachment and therefore no file name.

**Comment by jfarwer** [comment]
@chinuhub Yes, this happens to us. For example, one of our archives has about 280,000 attachments, and the report states there are about 8,000 attachments with no name. 

The error 'attachment with no filename' occurs when an email does not have a plain text part (text/plain) but just an HTML part (text/html). 

ePADD tries to recognise this situation by checking whether a text/html part is the top-level part of the message. If this is the case, it assumes there is no text/plain bit, extracts the text (removes the HTML tags) and uses that for displaying the email. However, it turns out the text/html part is not the top-level even if no text/plain exists, and therefore the extraction of the text does not happen at this point. What happens is that ePADD assumes the text/html is an HTML attachment. So the error message 'attachment with no filename' is added because the part is not an attachment and therefore has no filename. ePADD then extracts the text and saves it as email text. So in that way, the result is the same as correctly recognising the part as email text in the first place. However, to the report, the error message 'attachment with no filename' is added. The message also gets the label 'Error in Attachments (either this message has attachment(s) without file name or the parser is not able to fetch it)'.
It seems like all our cases, or at least almost all our cases of attachments with no filename relate to this issue. I will have a closer look.

I implemented a solution where a text/html part is recognised as email text when no text/plain comes in front. This assumes an existing text/plain part always comes before the text/html part. I am not sure whether this is always the case, but it works with the large email collections I used for testing.

I will do some more tests, and if everything works, make a pull request.
